internal/k8s/k8s.go

```diff
@@ -164,7 +165,7 @@ func New(cfg *Config) (*Client, error) {
 		c.syncFuncs = append(c.syncFuncs, c.svcInformer.HasSynced)
 
 		if cfg.ReadEndpoints {
-			if !UseEndpointSlices(c.client) {
+			if !UseEndpointSlices(c.client, cfg) {
```

Comment:
I'd change this to check the flag outside, seems more readable.

Something like `if cfg.DisableEPSlices || !UseEndpointSlices(c.client) {`

Reply:
Having all logic to determine if EndpointSlices should be used in one method seams like the reasonalbe thing to do, so I put it there. But I can change that.

Should I add a comment like I did in `UseEndpointSlices`? Or is it fine without that?

Reply:
Agree, but passing a flag just to return early won't add much value to the function itself I think. Might be different if we start using that in many different places, but that's not the current situation.

I'd move the comment too.

Reply:
Done